src/hal/stm32f4/pin.rs

```diff
@@ -151,6 +126,48 @@ impl PinConf {
   }
 }
 
+impl Gpio for Pin {
+    /// Sets output GPIO value to high.
+    fn set_high(&self) {
+      let bit: u32 = 1 << self.pin as usize;
+      self.get_reg().set_BSRR(bit);
+    }
+
+    /// Sets output GPIO value to low.
+    fn set_low(&self) {
+      let bit: u32 = 1 << (self.pin as usize + 16);
+      self.get_reg().set_BSRR(bit);
+    }
+
+    /// Returns input GPIO level.
+    fn level(&self) -> GpioLevel {
+      let bit: u32 = 1 << (self.pin as usize);
+      let reg = self.get_reg();
+
+      match reg.IDR() & bit {
+        0 => GpioLevel::Low,
+        _ => GpioLevel::High,
+      }
+    }
+
+    /// Sets output GPIO direction.
+    fn set_direction(&self, new_mode: GpioDirection) {
```

Comment:
You've got some 4-space indenting in here. Until we decide to go through and refactor the whole project (which will make `git blame` useless) we'll stick to the 2-space existing standard.